src/packages/admin-ui-components/src/filter-bar/component.tsx

```diff
@@ -1,129 +1,136 @@
-import { ReactNode, useEffect, useState, createElement } from 'react';
+import { ReactNode, useEffect, useState, createElement, useCallback } from 'react';
 import { useNavigate, useParams, useSearchParams } from 'react-router-dom';
+import toast from 'react-hot-toast';
+
 import { Button } from '../button';
+import { AdminUIFilterType, decodeSearchParams, Filter, routeFor, useSchema } from '../utils';
 import {
-	AdminUIFilterType,
-	decodeSearchParams,
-	FieldFilter,
-	Filter,
-	routeFor,
-	useSchema,
-} from '../utils';
-import {
+	BooleanFilter,
+	validateFilter,
 	DateRangeFilter,
-	DateRangeFilterType,
 	EnumFilter,
 	NumericFilter,
 	RelationshipFilter,
-	RelationshipFilterType,
 	TextFilter,
 } from '../filters';
 
 import styles from './styles.module.css';
-import { BooleanFilter } from '../filters/boolean-filter';
 
 export const FilterBar = ({ iconBefore }: { iconBefore?: ReactNode }) => {
-	const { entity, id } = useParams();
-	if (!entity) throw new Error('There should always be an entity at this point.');
+	const { entity: entityName, id } = useParams();
+	if (!entityName) throw new Error('There should always be an entity at this point.');
 	const [resetCount, setResetCount] = useState(0);
 	const [search] = useSearchParams();
 	const { entityByName } = useSchema();
 	const navigate = useNavigate();
 	const searchParams = decodeSearchParams(search);
-	const [filter, setFilter] = useState<FieldFilter>(
-		searchParams.filters ?? entityByName(entity).defaultFilter ?? {}
+	const [filters, setFilters] = useState(
+		searchParams.filters ?? entityByName(entityName).defaultFilter
 	);
 
-	if (!entity) {
-		throw Error('Entity should be in URL here');
-	}
-
-	const onFilter = (fieldName: string, filter?: Filter) => {
-		setFilter((_filter) => ({
-			..._filter,
-			...{ [fieldName]: filter },
-		}));
-	};
-
-	const getFilterComponents = (entityName: string) => {
-		const rowEntity = entityByName(entityName);
-
+	const filterFieldsOnEntity = useCallback(() => {
+		const entity = entityByName(entityName);
 		// @todo - currently the filters are not fitting on the screen
 		// we plan to redo this filter bar so that it is a drop down
 		// for now the workaround is to reduce the number of filters to 5
+		const showOnlyFiveFilters = entity.fields.length > 5 ? 5 : entity.fields.length;
+		const fields = entity.fields
+			// filter out rowEntity.fields with the JSON type
+			.filter((field) => field.type !== 'JSON')
+			.slice(0, showOnlyFiveFilters);
```

Comment:
Couldn't we simplify this to:
```typescript
const fields = entity.fields
	// filter out rowEntity.fields with the JSON type
	.filter((field) => field.type !== 'JSON')
	.slice(0, 5);
```
and just get rid of line 36? If not, it'd be like this I reckon:
```typescript
const fields = entity.fields
	// filter out rowEntity.fields with the JSON type
	.filter((field) => field.type !== 'JSON')
	.slice(0, Math.min(5, entity.fields.length));
```

Reply:
first one is good 👍